### ISSUE TITLE
Unexpected pre-commit behavior with v3.13.7

### ISSUE THREAD

**Issue by blink1073** [tcort/markdown-link-check#445]
Given the following two files in a git repo:

```bash
$ cat .pre-commit-config.yaml

repos:
- repo: https://github.com/tcort/markdown-link-check
  rev: v3.13.7
  hooks:
    - id: markdown-link-check
```

```bash
$ cat README.md

[bad link](does-not-exist.com)
```

The bad link is not caught by the current version:

```bash
$ pre-commit run --all-files markdown-link-check
Markdown Link Check......................................................Passed
```

If I change `.pre-commit-config.yaml` to use `3.12.2`:

```bash
$ pre-commit run --all-files markdown-link-check
Markdown Link Check......................................................Failed
- hook id: markdown-link-check
- exit code: 1

(node:34512) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

FILE: README.md
  [✖] does-not-exist.com

  1 links checked.

  ERROR: 1 dead links found!
  [✖] does-not-exist.com → Status: 400
````

**Comment by uvsmtid** [comment]
@blink1073, I believe the issue is that `markdown-link-check` exists with zero exit code: https://github.com/tcort/markdown-link-check/issues/463#issuecomment-2909599697

See if it is in your case as well.

**Comment by hofbi** [comment]
The 0 exit code issue has been reported in https://github.com/tcort/markdown-link-check/issues/457
